### DISCUSSION TITLE
Getting error as this error C2552: 'x509_w32_lookup' : non-aggregates cannot be initialized with initializer list with openssl 3.1.3

### DISCUSSION THREAD

**Discussion by pb1234** [openssl/openssl#23782] [Q&A] [answered]
our existing code is initializing as below which is working with old openssl, with new openssl getting the error as above:
static X509_LOOKUP_METHOD abc= 
{
	"test",
	aa,					
	bb,					
	NULL, 							
	NULL,							
	NULL,							
	cc,		
	NULL,  							
	dd,	
	NULL,							
};

**Comment by t8m** [comment, marked as answer]
Yes, this is expected. You have to use X509_LOOKUP_meth_new() and various X509_LOOKUP_meth_set_...() functions to set the lookup method members.

**Reply by nagu1137** [reply]
thanks for suggesting the right methods we agetting error when using X509_LOOKUP_meth_set_get_by_fingerprint.

 int x509_w32_get_by_fingerprint(X509_LOOKUP *ctx, X509_LOOKUP_TYPE type, unsigned char *bytes, int len, X509_OBJECT *ret)
 {
	
	

	return 1;
}

X509_LOOKUP_get_by_fingerprint_fn custom_get_by_fingerprint = static_cast<X509_LOOKUP_get_by_fingerprint_fn>(x509_w32_get_by_fingerprint);
int ret7=X509_LOOKUP_meth_set_get_by_fingerprint(x509_w32_lookup,custom_get_by_fingerprint);


 cannot convert from 'int (__cdecl *)(X509_LOOKUP *,X509_LOOKUP_TYPE,unsigned char *,int,X509_OBJECT *)' to 'X509_LOOKUP_get_by_fingerprint_fn'
28>        None of the functions with this name in scope match the target type

**Reply by nagu1137** [reply]
@t8m

**Reply by t8m** [reply]
It should be `const unsigned char *bytes` - you're missing the const qualifier.

**Reply by nagu1137** [reply]
Thank you

**Comment by pb1234** [comment]
Hi ,
our old code was using OpenSSL 1.0.2 q and it was using fips32, our code using these macros as below which were defined in fips32 of older version.
 CRYPTO_LOCK_SSL_CTX  
CRYPTO_LOCK_SSL_SESSION
CRYPTO_add(&a->references, 1, CRYPTO_LOCK_SSL_SESSION); //sample code for reference.

So now we updated to more recent version of OpenSSL 3.x, and these macros are not defined in fips32 now, but these are there in fips64, what can we do here, we tried just locally define in the source files but that is throwing linker errors.
